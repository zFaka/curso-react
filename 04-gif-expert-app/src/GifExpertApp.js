import React, {useState} from 'react';
import {AddCategory} from './components/atoms/AddCategory';
import {GifGrid} from './components/molecules/GifGrid';

export const GifExpertApp = ({defaultCategories=[]}) => {

  const [categories, setCategories] = useState(defaultCategories);

  //const handleAdd = () => {setCategories([...categories, 'Slime']);};

  return (
    <>
      <h2 className='titletitle'>GifExpertApp</h2> 
      <hr/>
      <AddCategory setCategories={setCategories}/>
      <ol>
        {
          categories.map(category => (
            <GifGrid 
              key={category}
              category={category}
            />
          ))
        }
      </ol>
    </>
  )}

export default GifExpertApp ;


