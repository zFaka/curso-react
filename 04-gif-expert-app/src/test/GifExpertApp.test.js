import React from 'react';
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('GifExpertApp', () => {

  test('Should show component', () => {

    const renderOutput = shallow(<GifExpertApp />);


    expect(renderOutput).toMatchSnapshot();
  })



  test('Should show a category list', () => {

    const categories = ['Jake', 'Finn'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);


    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  })



})
