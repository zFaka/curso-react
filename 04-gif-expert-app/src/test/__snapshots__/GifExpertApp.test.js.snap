// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Should show a category list 1`] = `
<Fragment>
  <h2
    className="titletitle"
  >
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Jake"
      key="Jake"
    />
    <GifGrid
      category="Finn"
      key="Finn"
    />
  </ol>
</Fragment>
`;

exports[`GifExpertApp Should show component 1`] = `
<Fragment>
  <h2
    className="titletitle"
  >
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
